gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaaas_driver.py

```diff
@@ -0,0 +1,418 @@
+            # failed. Return whichever agent is available.
+            for agent in agents:
+                if not agent['alive']:
+                    continue
+                return agent
+        except:
+            raise VPNAgentNotFound()
+
+        LOG.error(_('No active vpn agent found. Configuration will fail.'))
+        raise VPNAgentHostingServiceNotFound(vpnservice_id=vpnservice_id)
+
+    def _agent_notification(self, context, method, vpnservice_id,
+                            version=None, **kwargs):
+        """Notify update for the agent.
+            For some reason search with
+            'agent_type=AGENT_TYPE_VPN is not working.
+            Hence, get all the agents,
+            loop and find AGENT_TYPE_VPN, and also the one which
+            is hosting that vpn service and
+            implementing OC_IPSEC_TOPIC
+        """
+        admin_context = context.is_admin and context or context.elevated()
+
+        if not version:
+            version = self.target.version
+        vpn_agent = self._get_agent_hosting_vpnservice(
+            admin_context, vpnservice_id)
+
+        LOG.debug(_('Notify agent at %(topic)s.%(host)s the message '
+                    '%(method)s %(args)s'), {
+            'topic': self.topic, 'host': vpn_agent['host'],
+            'method': method, 'args': kwargs})
+
+        cctxt = self.client.prepare(server=vpn_agent['host'],
+                                    version=version)
+        cctxt.cast(context, method, **kwargs)
+
+    def vpnservice_updated(self, context, vpnservice_id, **kwargs):
+        """
+        Make rpc to agent for 'vpnservice_updated'
+        """
+        try:
+            self._agent_notification(
+                context, 'vpnservice_updated',
+                vpnservice_id, **kwargs)
+        except:
+            LOG.error(_('Notifying agent failed'))
+
+
+class OCIPsecVPNDriver(base_ipsec.BaseIPsecVPNDriver):
+    """VPN Service Driver class for IPsec."""
+
+    def __init__(self, service_plugin):
+        super(OCIPsecVPNDriver, self).__init__(
+            service_plugin,
+            OCVpnValidator(service_plugin))
+
+    def create_rpc_conn(self):
+        self._core_plugin = None
+        self.endpoints = [
+            OCIPsecVpnDriverCallBack(self),
+            agents_db.AgentExtRpcCallback(VPNPluginExt())]
+
+        self.conn = n_rpc.create_connection(new=True)
+        self.conn.create_consumer(
+            topics.VPN_PLUGIN_TOPIC, self.endpoints, fanout=False)
+        self.conn.consume_in_threads()
+        self.agent_rpc = OCIpsecVpnAgentApi(
+            topics.VPN_AGENT_TOPIC, BASE_IPSEC_VERSION, self)
+
+    @property
+    def service_type(self):
+        return 'VPN'
+
+    @property
+    def my_plugin(self):
+        return self.service_plugin
+
+    @property
+    def core_plugin(self):
+        if not self._core_plugin:
+            self._core_plugin = manager.NeutronManager.get_plugin()
+        return self._core_plugin
```

Comment:
This property core_plugin and self._core_plugin are unused.
They can be removed.